// react-github-calendar/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var Calendar = require('react-activity-calendar');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var Calendar__default = /*#__PURE__*/_interopDefaultLegacy(Calendar);

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

async function fetchCalendarData(username, year) {
  return (await fetch(`${API_URL}${username}?y=${year}`)).json();
}

const GitHubCalendar = ({
  username,
  year = 'last',
  theme = DEFAULT_THEME,
  ...calendarProps
}) => {
  const [data, setData] = React.useState(null);
  const [loading, setLoading] = React.useState(false);
  const [error, setError] = React.useState(null);
  const fetchData = React.useCallback(() => {
    setLoading(true);
    setError(null);
    fetchCalendarData(username, year).then(({
      contributions
    }) => setData(contributions)).catch(setError).finally(() => setLoading(false));
  }, [username, year]);
  React.useEffect(fetchData, []); // on mount

  React.useEffect(fetchData, [username, year]);

  if (error) {
    return /*#__PURE__*/React__default['default'].createElement("p", null, "Error :(");
  }

  if (loading || !data) {
    return /*#__PURE__*/React__default['default'].createElement(Calendar.Skeleton, {
      loading: true
    });
  }

  const labels = {
    totalCount: `{{count}} contributions in ${year === 'last' ? 'the last year' : '{{year}}'}`
  };
  return /*#__PURE__*/React__default['default'].createElement(Calendar__default['default'], _extends({
    data: data,
    theme: theme,
    labels: labels
  }, calendarProps));
}; // GitHub theme


const DEFAULT_THEME = {
  level4: '#216e39',
  level3: '#30a14e',
  level2: '#40c463',
  level1: '#9be9a8',
  level0: '#ebedf0'
};
const API_URL = 'https://github-contributions-api.jogruber.de/v4/';

exports['default'] = GitHubCalendar;
